dids/diddht/bep44.go

```diff
@@ -0,0 +1,95 @@
+package diddht
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/anacrolix/torrent/bencode"
+)
+
+// Represents a BEP44 message, which is used for storing and retrieving data in the Mainline DHT
+// network.
+//
+// A BEP44 message is used primarily in the context of the DID DHT method for publishing and
+// resolving DID documents in the DHT network. This type encapsulates the data structure required
+// for such operations in accordance with BEP44.
+//
+// https://www.bittorrent.org/beps/bep_0044.html
+type bep44Message struct {
+
+	// The public key bytes of the Identity Key, which serves as the identifier in the DHT network for
+	// the corresponding BEP44 message.
+	//
+	k []byte
+
+	// The sequence number of the message, used to ensure the latest version of the data is retrieved
+	// and updated. It's a monotonically increasing number.
+	seq int64
+
+	// The signature of the message, ensuring the authenticity and integrity of the data. It's
+	// computed over the bencoded sequence number and value.
+	sig []byte
+
+	// The actual data being stored or retrieved from the DHT network, typically encoded in a format
+	// suitable for DNS packet representation of a DID Document.
+	v []byte
+}
+
+type signer func(payload []byte) ([]byte, error)
+
+// newSignedBEP44Message creates a new signed BEP44 message with the given DNS payload.
+func newSignedBEP44Message(dnsPayload []byte, seq int64, publicKeyBytes []byte, signer signer) (*bep44Message, error) {
```

Comment:
Total golang n00b question: if only names starting with capital-letters are exported, and this func doesn't seem to be used outside of tests, what's the point? is the intention to use it in this package in a follow up PR?

Reply:
correct. I will use it in the did create that's in a draft PR. I don't want it exported just yet since this is mostly a DHT internal implementation detail. I'm open to arguments otherwise though

Reply:
no pushback from me, was just curious. ty!